chsdi/response_callbacks.py

```diff
@@ -11,6 +11,17 @@ def add_default_cache_control(request, response):
         dft_cache_control = request.registry.settings['default_cache_control']
         response.cache_control = CacheControl.parse(dft_cache_control, type='response')
 
+    # overwrite with these 4xx/5xx cache settings
+    # no cache on these 5xx errors, they are supposed to be temporary
+    if response.status_code in (502, 503, 504, 507):
+        response.headers['Cache-Control'] = 'no-cache'
+    # short cache duration for other 5xx errors
+    elif response.status_code >= 500:
+        response.headers['Cache-Control'] = 'public, max-age=10'
```

Comment:
In `pyramid` you cannot directly set the Cache-Control header it will not work as expected, you need to do it via `cache_control` property see line 12.